AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/FocusVisualizer.java

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package com.microsoft.accessibilityinsightsforandroidservice;
+
+import android.content.Context;
+import android.view.accessibility.AccessibilityEvent;
+import android.view.accessibility.AccessibilityNodeInfo;
+import java.util.ArrayList;
+
+public class FocusVisualizer {
+  private ArrayList<FocusElementHighlight> focusElementHighlights;
+  private ArrayList<FocusElementLine> focusElementLines;
+  private int tabStopCount;
+  private Context context;
+  private FocusVisualizerStyles styles;
+  private FocusCanvasView focusCanvasView;
+
+  public FocusVisualizer(
+      Context context,
+      FocusVisualizerStyles focusVisualizerStyles,
+      FocusCanvasView focusCanvasView) {
+    this.context = context;
+    this.focusElementHighlights = new ArrayList<>();
+    this.focusElementLines = new ArrayList<>();
+    this.tabStopCount = 0;
+    this.styles = focusVisualizerStyles;
+    this.focusCanvasView = focusCanvasView;
+  }
+
+  public void HandleAccessibilityRedrawEvent(AccessibilityEvent event) {}
+
+  public void HandleAccessibilityFocusEvent(AccessibilityEvent event) {
+    tabStopCount++;
+
+    AccessibilityNodeInfo eventSource = event.getSource();
+    AccessibilityNodeInfo previousEventSource = this.getPreviousEventSource();
+
+    this.createFocusElementHighlight(eventSource);
+    this.createFocusElementLine(eventSource, previousEventSource);
+
+    if (this.focusElementHighlights.size() > 1) {
+      this.setPreviousElementHighlightNonCurrent(
+          this.focusElementHighlights.get(this.focusElementHighlights.size() - 2));
+    }
+    if (focusElementLines.size() > 1) {
+      this.setPreviousLineNonCurrent(this.focusElementLines.get(this.focusElementLines.size() - 2));
+    }
+
+    this.focusCanvasView.setFocusElementHighlights(this.focusElementHighlights);
```

Comment:
nit: seems like this can just be one method for setting the focus highlights/lines.

Reply:
updated to a single method